src/backend/joanie/core/api/client.py

```diff
@@ -357,42 +350,34 @@ def create(self, request, *args, **kwargs):
                 status=400,
             )
 
-        # Once order has been created, if product is not free, create a payment
-        if product.price > 0:
-            order = serializer.instance
-            payment_backend = get_payment_backend()
-            credit_card_id = serializer.initial_data.get("credit_card_id")
-
-            # if payment in one click
-            if credit_card_id:
-                try:
-                    credit_card = CreditCard.objects.get(
-                        owner=order.owner, id=credit_card_id
-                    )
-                    payment_info = payment_backend.create_one_click_payment(
-                        request=request,
-                        order=order,
-                        billing_address=billing_address,
-                        credit_card_token=credit_card.token,
-                    )
-                except (CreditCard.DoesNotExist, NotImplementedError):
-                    pass
-            else:
-                payment_info = payment_backend.create_payment(
-                    request=request, order=order, billing_address=billing_address
-                )
+        # Else return the fresh new order
+        return Response(serializer.data, status=201)
 
-            # Return the fresh new order with payment_info
+    @action(detail=True, methods=["PUT"])
```

Comment:
Could be a patch nope ? In order to provide only required information to transition to submit state ? (`billing_address`, `order_id` and `credit_card_id`) ?
```suggestion
    @action(detail=True, methods=["PATCH"])
```